### PR TITLE
Add support Chef 15

### LINKED ISSUE #204: [BUG] Xcode install fails on Chef 15
## Describe the Bug 

`xcode` resource fails to install Xcode on Chef 15 due to `gmkdir` missing.

## Error Message  

```bash
STDERR: ERROR: Error installing xcode-install: ERROR: Failed to build gem native extension.
```

## Chef Exception  

```bash
Running handlers:
2205
[2019-07-08T12:12:40-07:00] ERROR: Running exception handlers
2206
Running handlers complete
2207
[2019-07-08T12:12:40-07:00] ERROR: Exception handlers complete
2208
Chef Infra Client failed. 224 resources updated in 11 minutes 59 seconds
2209
[2019-07-08T12:12:40-07:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
2210
[2019-07-08T12:12:40-07:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
2211
[2019-07-08T12:12:40-07:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: xcode[install Xcode 9.4.1] (apex_automation::xcode line 6) had an error: Mixlib::ShellOut::ShellCommandFailed: chef_gem[xcode-install] (/tmp/kitchen/cache/cookbooks/macos/resources/xcode.rb line 12) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
2212
---- Begin output of /opt/chef/embedded/bin/gem install xcode-install -q --no-document -v "2.5.0" --source=https://www.rubygems.org --no-document --no-user-install ----
2213
STDOUT: Successfully installed claide-1.0.2
2214
Successfully installed slack-notifier-2.3.2
2215
Successfully installed atomos-0.1.3
2216
Successfully installed CFPropertyList-3.0.0
2217
Successfully installed colored2-3.1.2
2218
Successfully installed nanaimo-0.2.6
2219
Successfully installed xcodeproj-1.10.0
2220
Successfully installed rouge-2.0.7
2221
Successfully installed xcpretty-0.3.0
2222
Successfully installed terminal-notifier-2.0.0
2223
Successfully installed terminal-table-1.8.0
2224
Successfully installed public_suffix-2.0.5
2225
Successfully installed multipart-post-2.0.0
2226
Successfully installed word_wrap-1.0.0
2227
Successfully installed tty-cursor-0.7.0
2228
Successfully installed tty-spinner-0.9.1
2229
Successfully installed babosa-1.0.2
2230
Successfully installed colored-1.2
2231
Successfully installed commander-fastlane-4.4.6
2232
Successfully installed excon-0.64.0
2233
Building native extensions. This could take a while...
2234
STDERR: ERROR: Error installing xcode-install:
2235
ERROR: Failed to build gem native extension.
2236
2237
current directory: /opt/chef/embedded/lib/ruby/gems/2.6.0/gems/unf_ext-0.0.7.6/ext/unf_ext
2238
/opt/chef/embedded/bin/ruby -I /opt/chef/embedded/lib/ruby/site_ruby/2.6.0 -r ./siteconf20190708-27933-15k8xci.rb extconf.rb
2239
checking for -lstdc++... yes
2240
creating Makefile
2241
2242
current directory: /opt/chef/embedded/lib/ruby/gems/2.6.0/gems/unf_ext-0.0.7.6/ext/unf_ext
2243
make "DESTDIR=" clean
2244
2245
current directory: /opt/chef/embedded/lib/ruby/gems/2.6.0/gems/unf_ext-0.0.7.6/ext/unf_ext
2246
make "DESTDIR="
2247
compiling unf.cc
2248
linking shared-object unf_ext.bundle
2249
2250
current directory: /opt/chef/embedded/lib/ruby/gems/2.6.0/gems/unf_ext-0.0.7.6/ext/unf_ext
2251
make "DESTDIR=" install
2252
make: /usr/local/bin/gmkdir: No such file or directory
2253
make: *** [.sitearchdir.time] Error 1
2254
2255
make install failed, exit code 2
```

Related to AB#3477094

### PR DESCRIPTION
**Pull request by ehanlon**
- Added suites for Chef 15 to the Azure Pipelines strategy
- Fixes #204

### CODE REVIEW

File: spec/unit/resources/xcode_spec.rb [hunk 1]
```diff
@@ -12,21 +12,6 @@
       .and_return(true)
     allow_any_instance_of(MacOS::CommandLineTools).to receive(:macos_version)
```
Comment (jazaval) [inline spec/unit/resources/xcode_spec.rb]: You can remove this stub as well.

File: test/integration/default/controls/xcode_test.rb [hunk 2]
```diff
@@ -10,18 +10,3 @@
     it { should_not be_symlink }
   end
 end
-
-control 'command-line-tool-sentinel' do
```
Comment (jazaval) [inline test/integration/default/controls/xcode_test.rb]: I would leave this to test against any regressions - as we probably won't use the gem workaround forever.

Comment (ehanlon) [reply test/integration/default/controls/xcode_test.rb]: This workaround was documented [3 years ago](https://github.com/xcpretty/xcode-install/commit/a00f22ea68ad924eaa667b3236c0f75c9054bfd7), and the timeline for a permanent fix seems indefinite.

We do have a `command_line_tools` resource in the cookbook, and perhaps this should be a separate unit test for that resource specifically?

Comment (jazaval) [reply test/integration/default/controls/xcode_test.rb]: At some point we will resume installing command line tools via some method yet to be determined, but all of those methods rely on touching this file. This test ensures that the method of installation properly removes the file - if there are tests in that specific tooling for this file then I'd say it's fine to remove.

Comment (ehanlon) [reply test/integration/default/controls/xcode_test.rb]: At this point, we can add separate tests for the `command_line_tools` resource.

Going forward, we need to upstream the tests to `build_essential`, and remove the duplication from our resource cookbook. Related to #206

**Pull request merged**